### ISSUE TITLE
Check for stale PID files (better cron support)

### ISSUE THREAD

**Issue by Ryan-Goldstein** [GLolol/PyLink#512]
It'd be great to have cronjob support for PyLink to start it automatically in the case of a crash, system reboot, etc.

I recommend adding something like UnrealIRCd's 'croncheck' flag:
```bash
elif [ "$1" = "croncheck" ] ; then
        if [ -r $PID_FILE ] ; then
                kill -CHLD `cat $PID_FILE` 1>/dev/null 2>&1
                if [ "$?" = 0 ]; then
                        # IRCd is running, bail out silently.
                        exit 0
                fi
        fi
        # PID file not found or found but stale
        echo "UnrealIRCd is not running. Starting now..."
        $0 start
```

Reference: https://github.com/unrealircd/unrealircd/blob/unreal40/unrealircd.in#L94

**Comment by GLolol** [comment]
Run `pylink` with the `-c` argument in a cronjob if you're on 1.x (this is implied on 2.x/devel). If it finds a PID file already present, it will quit automatically. Otherwise, it'll start the service.

**Comment by Ryan-Goldstein** [comment]
It would be good to check whether the PID specified in the .pid file is running, silently exiting if so. Simply checking for the existence of the PID file means that any unexpected termination, like a crash or unexpected system reboot, won't permit automatic launching via cron.

**Comment by GLolol** [comment]
Implemented in 2.0-dev, see above commit. You will need to install `psutil` for the process existence checks to work; otherwise it will still just check whether the PID file exists.
